packages/ai/core/generate-text/stream-text.ts

```diff
@@ -1541,7 +1541,7 @@ However, the LLM results are expected to be small enough to not cause issues.
     }
 
     const state = createStreamingUIMessageState({
-      lastMessage,
+      lastMessage: structuredClone(lastMessage),
```

Comment:
Moved `structuredClone` of last message out of `createStreamingUIMessageState` (explanation in corresponding file)